### PR TITLE
[image_picker] Fix NSNull handling on iOS

### PR DESCRIPTION
**Pull request by stuartmorgan**
Instead of assigning NSNull to NSNumber ivars, which is a type violation
that requires confusing and error-prone code elsewhere, convert to nil
before assignment.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
This was making me sad every time I reviewed the open PR that touches code doing one of these checks.

**Review by jmagman — APPROVED** [review]
Yikes, thanks

File: packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m [hunk 1]
```diff
@@ -16,6 +16,15 @@
...
 #import "FLTPHPickerSaveImageToPathOperation.h"
 
+/**
+ * Returns the value for the given key in 'dict', or nil if the value is
+ * NSNull.
+ */
+NSObject *GetNullableValueForKey(NSDictionary *dict, NSString *key) {
+  NSObject *value = dict[key];
+  return value == [NSNull null] ? nil : value;
+}
```
Comment (jmagman) [inline packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:26]: Actually the analyzer doesn't like that this is a `NSObject`.
```
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:409:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *maxWidth = GetNullableValueForKey(self->_arguments, @"maxWidth");
              ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:410:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *maxHeight = GetNullableValueForKey(self->_arguments, @"maxHeight");
              ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:411:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *imageQuality = GetNullableValueForKey(self->_arguments, @"imageQuality");
              ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:492:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *maxWidth = GetNullableValueForKey(_arguments, @"maxWidth");
              ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:493:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *maxHeight = GetNullableValueForKey(_arguments, @"maxHeight");
              ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/folders/tn/f_9sf1xx5t14qm_6f83q3b840000gn/T/cirrus-ci-build/packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:494:15: error: incompatible pointer types initializing 'NSNumber *' with an expression of type 'NSObject *' [-Werror,-Wincompatible-pointer-types]
    NSNumber *imageQuality = GetNullableValueForKey(_arguments, @"imageQuality");
              ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```
```suggestion
id GetNullableValueForKey(NSDictionary *dict, NSString *key) {
  id value = dict[key];
  return value == [NSNull null] ? nil : value;
}
```

Comment (jmagman) [reply packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m:26]: (woo analyzer running in presubmit!)

Comment (stuartmorgan) [reply packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m]: I thought that might happen but couldn't remember how NSObject* was handled. I meant to check locally before pushing 😛 

Fixed! Also fixed the "non-nullable" arguments on `scaledImage:...` that had handling for `NSNull` to be nullable now that they are actually nil.

**Pull request merged**